content/articles/understanding-polymorphism-abstract-classes-using-c-/index.md

```diff
@@ -0,0 +1,327 @@
+According to [Microsoft](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/object-oriented/polymorphism), **Polymorphism** is often referred to as the third pillar of object-oriented programming after encapsulation and inheritance. *Polymorphism* is a Greek word that means "many-shaped". At run time, objects of a derived class may be treated as objects of a base class in places such as method parameters and collections or arrays. When this polymorphism occurs, the object's declared type is no longer identical to its run-time type.
+
+**Polymorphism** is a very helpful concept when it comes to a single abstract idea used in different ways and shapes. With polymorphism, you can define a single abstract class and reshape it in different ways to suit your methods & ideas. 
+
+### Table of contents
+
+- Understand the concept of polymorphism.
+- Understand abstract classes.
+- Multiple examples of polymorphism.
+
+### Prerequisites
+
+To follow this article along, you'll need the following:
+
+- A basic understanding of the C# programming language.
+- A basic understanding of Inheritance in C# programming language.
+
+If you're new to Inheritance, you should check this tutorial that would help you understand the concept of it [here](https://www.section.io/engineering-education/getting-started-with-inheritance-using-c/).
+
+### Abstract classes
+
+Abstract classes are too generic to be instantiated and are used as base classes. The definitions for abstract classes are not complete. The derived classes must be defined as the missing pieces and should contain abstract methods or abstract properties.
+
+The following example will explain how we can define an abstract class to make a Payroll System Using Polymorphism.

```

Comment:
It seems like you can provide a detailed explanation of what abstract classes & abstract methods are before jumping into the example.